src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx

```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
     [openTrade, trade],
   );
 
+  const margin = useMemo(
+    () =>
+      trade
+        ? getRequiredMarginCollateral(
+            trade!.leverage,
+            marginAccountBalance.fPositionBC,
+            Number(trade!.amount),
+            perpParameters,
+          )
+        : 0,
+    [marginAccountBalance.fPositionBC, perpParameters, trade],
+  );
+
+  const tradingFee = useMemo(
+    () => (trade ? getTradingFee(Number(trade!.amount), perpParameters) : 0),
+    [perpParameters, trade],
+  );
+
+  const isBuy = useMemo(() => trade?.position === TradingPosition.LONG, [
+    trade?.position,
+  ]);
+
+  if (!trade) {
+    return null;
+  }
+
   return (
     <Dialog
       isOpen={modal === PerpetualPageModals.TRADE_REVIEW}
       onClose={onClose}
     >
       <h1>{t(translations.perpetualPage.reviewTrade.title)}</h1>
-      {/* TODO: implement Review Trade Dialog */}
-      <pre>{JSON.stringify(trade, null, 2)}</pre>
-      <button onClick={onSubmit}>Open trade</button>
+      <div className={styles.contentWrapper}>
+        <div className={styles.importantInfo}>
+          <div
+            className={classNames(
+              'tw-text-xl tw-font-medium tw-tracking-normal tw-leading-none',
+              isBuy ? styles.orderActionBuy : styles.orderActionSell,
+            )}
+          >
+            {trade?.leverage}x{' '}
+            {t(translations.perpetualPage.reviewTrade.market)}{' '}
+            {getTradePosition(trade?.position)}
+          </div>
+          <div className={styles.orderActionAdditionalInfo}>
+            {toNumberFormat(fromWei(trade?.amount), 3)} BTC @{' '}
+            {isBuy ? '≥' : '≤'} {toNumberFormat(trade.entryPrice, 2)} USD
```

Comment:
Please use the pair.baseAsset and pair.qouteAsset. If you took this from Trade Form please update it there too.